lib/collection/src/operations/universal_query/collection_query.rs

```diff
@@ -20,7 +20,7 @@ use crate::operations::types::{CollectionError, CollectionResult};
 use crate::recommendations::avg_vector_for_recommendation;
 
 /// Internal representation of a query request, used to converge from REST and gRPC. This can have IDs referencing vectors.
-#[derive(Debug)]
+#[derive(Clone, Debug, PartialEq)]
```

Comment:
`Clone` + `PartialEq` is a requirement of our test logic for grouping, not for actual runtime.

Reply:
If you would like to [compile it conditionally](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg_attr-attribute), it would look something like this
```suggestion
#[derive(Debug)]
#[cfg_attr(test, derive(Clone, PartialEq)]
```

But I think the extra derives are small enough to not even bother